src/index.js

```diff
@@ -1,32 +1,77 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+// const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+function fetchData(url_api) {
+
+  return new Promise((resolve, reject) => {
+
+    // console.log("Entro a promise...")
+    let xhttp = new XMLHttpRequest();
+    xhttp.open('GET', url_api, true); 
+    
+    //async: true; El event loop NO se bloquea y no aparece el error de main thread bloqueado
+    //async: false; El main thread se bloqueará
+
+    xhttp.onreadystatechange = () => {
+      // console.log(xhttp.readyState)
+      if (xhttp.readyState == 4) {
+        if (xhttp.status == 200) {
+          // console.log("resolvio el callback")
+          return resolve(JSON.parse(xhttp.responseText));
+        }
+        // console.log("rejected")
+        // return resolve(JSON.parse(xhr.responseText));
+        return reject(url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```